pkgs/development/libraries/libsmu/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, pkg-config
+, libusb1
+, boost
+, doxygen
+, pythonSupport ? false
+, python
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libsmu";
+  version = "1.0.3";
+
+  src = fetchFromGitHub {
+    owner = "analogdevicesinc";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0m3bfrs1ipv2mvfhvfsmaw30xl5f7pdw70pd5ahj3lgiyfss74z3";
+  };
+
+  nativeBuildInputs = with python.pkgs; [
+    cmake
+    pkg-config
+    doxygen
+  ]
+  ++ lib.optionals (pythonSupport) [
+    python
+    cython
+    python.pkgs.setuptools
+  ]
+  ;
```

Comment:
The formatting here is all kinds of weird. It's probably easier for you to just run `nixpkgs-fmt` on this rather than me direct you the precise changes.

Reply:
sorry for the late reply, github always fails to update my open tab...

i think i know why you find it odd
i know why i wrote it like that
and nixpkgs-fmt agrees with me
`0 / 1 would have been reformatted`

i like separating the semicolon when joining lists, it makes for easier per-line editing and neater diffs
if there was a style guide for nixpkgs i'd be willing to follow that, but lacking that, i'd like to stick to what i like

Reply:
hmm, the redundant python.pkgs for setuptools is odd though, fixed that

Reply:
nixpkgs-fmt does not generate the same output for every input.